File: OneCodeTeam/Control-IIS-in-three-diffrent-ways-in-Azure-22782/[VB]-Control IIS in three diffrent ways in Azure/VB/VBAuzreControlIIS/ControlIIS/Controllers/HomeController.vb
'****************************** Module Header ******************************\
' Module Name:  HomeController.vb
' Project:     ControlIIS
' Copyright (c) Microsoft Corporation.
' 
' As we know, cloud service can full control IIS. If someone want to change IIS, 
' they will probably first think about using startup script, since it has been 
' documented in Azure training kit.
' That's a good way but not the only way.
' Actually we can use site class and Configuration config to achieve that.
' This sample will show you how can we control IIS by two different ways in Azure
' Cloud service.
' 
' This class will show you some good ways
' 
' This source is subject to the Microsoft Public License.
' See http://www.microsoft.com/en-us/openness/licenses.aspx#MPL
' All other rights reserved.
' 
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
' EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
' WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
' ***************************************************************************/
Imports Microsoft.Web.Administration
Imports Microsoft.WindowsAzure.ServiceRuntime

Public Class HomeController
    Inherits System.Web.Mvc.Controller

    '
    ' GET: /Home
    Public Function Index() As ActionResult

        Return View()
    End Function

    Public Function ShowDetails() As ActionResult
        Using manager As New ServerManager()
            Dim webSite As Site = manager.Sites(RoleEnvironment.CurrentRoleInstance.Id + "_Web")

            Dim config As Configuration = manager.GetApplicationHostConfiguration()
            Dim AppPoolName As String = webSite.Applications(0).ApplicationPoolName

            Dim applicationPoolsSection As ConfigurationSection = config.GetSection("system.applicationHost/applicationPools")
            Dim applicationPoolsCollection As ConfigurationElementCollection = applicationPoolsSection.GetCollection()
            Dim addElement As ConfigurationElement = ConfigurationUtiltiy.FindElement(applicationPoolsCollection, "add", "name", AppPoolName)
            'Get Current Site server's Pipeline mode by configurationSection.
            If addElement Is Nothing Then
                Throw New InvalidOperationException("Element not found!")
            End If
            If addElement("managedPipelineMode").ToString() = "0" Then
                ViewBag.CurrentServerPipelineMode = "Intergrate"
            Else
                ViewBag.CurrentServerPipelineMode = "Classic"
            End If
            'Get the site name on instance IIS.
            ViewBag.ServerName = manager.Sites(RoleEnvironment.CurrentRoleInstance.Id + "_Web").Name
        End Using
        Return View()
    End Function



End Class

File: OneCodeTeam/Control-IIS-in-three-diffrent-ways-in-Azure-22783/[C#]-Control IIS in three diffrent ways in Azure/C#/CSAuzreControlIIS/ControlIIS/Controllers/HomeController.cs
/****************************** Module Header ******************************\
Module Name:  HomeController.cs
Project:     ControlIIS
Copyright (c) Microsoft Corporation.
 
As we know, cloud service can full control IIS. If someone want to change IIS, 
they will probably first think about using startup script, since it has been 
documented in Azure training kit.
That's a good way but not the only way.
Actually we can use site class and Configuration config to achieve that.
This sample will show you how can we control IIS by two different ways in Azure
Cloud service.

This class will show you some good ways
 
This source is subject to the Microsoft Public License.
See http://www.microsoft.com/en-us/openness/licenses.aspx#MPL
All other rights reserved.
 
THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
\***************************************************************************/

using Microsoft.Web.Administration;
using Microsoft.WindowsAzure.ServiceRuntime;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using ControlIIS;

namespace ControlIIS.Controllers
{
    public class HomeController : Controller
    {
        //
        // GET: /Home/

        public ActionResult Index()
        {
          
            return View();
        }

        public ActionResult ShowDetails()
        {
            using (ServerManager manager = new ServerManager())
            {
                Site webSite = manager.Sites[RoleEnvironment.CurrentRoleInstance.Id + "_Web"];

                Configuration config = manager.GetApplicationHostConfiguration();
                string AppPoolName = webSite.Applications[0].ApplicationPoolName;

                ConfigurationSection applicationPoolsSection = config.GetSection("system.applicationHost/applicationPools");
                ConfigurationElementCollection applicationPoolsCollection = applicationPoolsSection.GetCollection();
                ConfigurationElement addElement = ConfigurationUtiltiy.FindElement(applicationPoolsCollection, "add", "name", AppPoolName);

                // Get Current Site server's Pipeline mode by configurationSection.
                if (addElement == null) throw new InvalidOperationException("Element not found!");
                if (addElement["managedPipelineMode"].ToString() == "0")
                    ViewBag.CurrentServerPipelineMode = "Intergrate";
                else
                    ViewBag.CurrentServerPipelineMode = "Classic";

                // Get the site name on instance IIS.
                ViewBag.ServerName = manager.Sites[RoleEnvironment.CurrentRoleInstance.Id + "_Web"].Name;   
            }
            return View();
        }

    }
}
